### ISSUE TITLE
[Feature] Contract Tagging

### ISSUE THREAD

**Issue by 0xphilipp** [alleslabs/celatone-frontend#440]
Reasoning about what happens exactly in the Event Logs is difficult if you dont know what an address or wallet is.

There should be 2 or 3 ways of tagging.

1. Global tagging managed by celatone / community that will be default for everyone. (Simple submission of wallet tags is necessary to make the experience smooth, ideally through the page)
2. User tagging - Users should be able to tag their own addresses and to be better to reason about it.
3. Optional: Index the labels of the contract instantiations and use them as default tag shown in the events log / guis. This should be able to be set based on a user setting, as it could also have some misleading / bad data on instantiations.

If a tag available instead of the full address, just show: {TAG} (neutron1...mkd2)

![image](https://github.com/alleslabs/celatone-frontend/assets/9350951/4bb59982-5495-4855-bb57-47868c243443)

**Comment by tansawit** [comment]
Definitely! It's something we will be working on. Though it will require quite a bit of revamp on how we're currently parsing and sending data to the frontend to visualize, and also (currently) lower priority than things like contract verification/JSON Schema/state explorer that we've been getting more demand fo. So will be a bit before this gets to production.
